python/triton/runtime/interpreter.py

```diff
@@ -1,32 +1,36 @@
+from __future__ import annotations
 import ast
 import textwrap
 import inspect
-from typing import Tuple, List
+from typing import Tuple, List, Dict
 
 import math
 import numpy as np
 
 import triton
 import triton.language as tl
+import dataclasses
 from dataclasses import dataclass
+
+from triton.language import semantic
+from triton.tools.experimental_descriptor import TensorDescriptor
```

Comment:
Still experimental?

Reply:
I don't think the API will change I just haven't decided where to put the class yet.